test/e2e/specs/editor/blocks/navigation.spec.js

```diff
@@ -48,5 +60,89 @@ test.describe(
 <!-- /wp:navigation -->`
 			);
 		} );
+
+		test( 'default to my only existing menu', async ( {
+			editor,
+			navBlockUtils,
+		} ) => {
+			const createdMenu = await navBlockUtils.createNavigationMenu( {
+				title: 'Test Menu 1',
+				content:
+					'<!-- wp:navigation-link {"label":"WordPress","type":"custom","url":"http://www.wordpress.org/","kind":"custom","isTopLevelLink":true} /-->',
+			} );
+
+			await editor.insertBlock( { name: 'core/navigation' } );
+
+			// Check the markup of the block is correct.
+			await editor.publishPost();
+			const content = await editor.getEditedPostContent();
+			expect( content ).toBe(
+				`<!-- wp:navigation {"ref":${ createdMenu.id }} /-->`
+			);
```

Comment:
Nit: We can also leverage `editor.getBlocks()` as shown [here](https://github.com/WordPress/gutenberg/blob/trunk/docs/contributors/code/e2e/overusing-snapshots.md#the-solution) for slightly more readable output.

```suggestion
			await expect.poll( editor.getBlocks ).toEqual( [ {
				name: 'core/navigation',
				ref: createdMenu.id,
			} ] );
```

Reply:
thank you, today I learned!

Reply:
I'm getting this:
```

    - Expected  -  1
    + Received  + 13

      Array [
        Object {
    -     "name": "core/navigation",
    +     "attributes": Object {
    +       "hasIcon": true,
    +       "icon": "handle",
    +       "maxNestingLevel": 5,
    +       "openSubmenusOnClick": false,
    +       "orientation": "horizontal",
    +       "overlayMenu": "mobile",
            "ref": 127,
    +       "showSubmenuIcon": true,
    +     },
    +     "clientId": "ded86be9-a2b7-4318-8e8f-d79089a0eaff",
    +     "innerBlocks": Array [],
    +     "isValid": true,
    +     "name": "core/navigation",
        },
      ]
```

Reply:
Oh, right. It matches the full block's content. Instead, we should use `toMatch`:

```js
await expect.poll( editor.getBlocks ).toMatch( [ {
	name: 'core/navigation',
	attributes: { ref: createdMenu.id },
} ] );
```

Reply:
`toMatchObject` did the trick, `toMatch` was expecting a string 👍